[drustanyjt] iP

drustanyjt opened the pull request:
## Pull Request

### Description

This PR addresses the following changes:

- Updated error messages for clarity and conciseness.
- Added `Command` enum for better command handling.
- Implemented auto-save functionality to persist tasks.
- Refactored error handling in various methods.
- Added concise DukeException messages for improved user experience.

### Changes Made

1. Modified error messages in several methods for improved clarity and brevity.
2. Introduced `Command` enum to facilitate command handling and processing.
3. Implemented auto-save functionality to persist tasks to a file.
4. Refactored error handling logic in relevant methods for better maintainability and readability.
5. Updated DukeException messages to be concise and informative for users.

### Checklist

- [x] Updated error messages
- [x] Implemented `Command` enum
- [x] Added auto-save functionality
- [x] Refactored error handling logic
- [x] Updated DukeException messages

### Sample Code

The main **entry point** for the application is the `Duke` class.

```
public class Duke{

    /**
     * Main method to run the Duke program.
     * @param args The command line arguments.
     */
    public static void main(String[] args) {
        Application.launch(DukeApp.class);
    }
}

```

And commands are generally executed through `c.execute(tasks, ui, storage)` within `DukeApp`.

### Additional Notes

**Important:** Make sure all changes are reviewed and tested before merging.

:rocket: Let's make Fluffy even better!

> "Great things never came from comfort zones."

For more details, check out the [documentation](https://github.com/drustanyjt/ip).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

drustanyjt committed:
changed name to fluffy

drustanyjt committed:
added echo

drustanyjt committed:
refactor while loop

drustanyjt committed:
added mark and unmark commands and Task class

drustanyjt committed:
added 3 task subclasses

drustanyjt committed:
refactored with ChatGPT to make main start an INSTANCE of Duke to handle chats

drustanyjt committed:
added delete and fixed exceptions not clearing the input buffer

drustanyjt committed:
added enums and fixed scanner (also making commands case insensitive)

drustanyjt committed:
added auto save

drustanyjt committed:
Merge branch 'branch-level-7'

drustanyjt committed:
Add date time logic to Task subclasses

The constructors for both Event and Deadline now convert
date strings into actual LocalDate objects.

drustanyjt committed:
Merge branch 'branch-level-8'

drustanyjt committed:
Refactor code into various classes for clearer abstractions

drustanyjt committed:
Merge branch 'A-MoreOOP'

drustanyjt committed:
Group java source files into logical packages

drustanyjt committed:
Merge branch 'branch-a-packages'

drustanyjt committed:
Merge branch 'add-gradle-support' into branch-a-gradle

drustanyjt committed:
Fix entry point to be in the duke package's Duke class

drustanyjt committed:
Merge branch 'branch-a-gradle'

drustanyjt committed:
Make Parser methods package-accessible and JUnit tests for some methods

drustanyjt committed:
Merge branch 'branch-a-junit'

drustanyjt committed:
Add JavaDocs using copilot

drustanyjt committed:
Merge branch 'branch-A-JavaDoc'

drustanyjt committed:
Fix line breaks to appear after operators as per style guide

drustanyjt committed:
Merge branch 'branch-A-CodingStandard'

drustanyjt committed:
Add Find command

drustanyjt committed:
Merge branch 'branch-level-9'

SherisseTJW commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+     * @param args The command line arguments.
+     */
+    public static void main(String[] args) {
+        new Duke("data/duke.txt").run();
```
Perhaps you can have the filepath as a `private final` field so that it is more easily accessible and changed?

SherisseTJW commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,60 @@
+     * @param fileString The file string to be converted to a todo.
+     * @return The todo from the file string.
+     */
+    public static Todo TodoFromFileString(String fileString) {
```
```suggestion
    public static Todo getTodoFromFileString(String fileString) {
```

Perhaps rename it like so? Note that methods have to be verbs and in camel case

SherisseTJW commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,87 @@
+     * @param fileString The file string to be converted.
+     * @return The Deadline object from the file string.
+     */
+    public static Deadline DeadlineFromFileString(String fileString) {
```
Perhaps rename it like so? Note that methods have to be verbs and in camel case
```suggestion
    public static Deadline getDeadlineFromFileString(String fileString) {
```

SherisseTJW commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,94 @@
+     * @param fileString The file string to be converted to an event.
+     * @return The event from the file string.
+     */
+    public static Event EventFromFileString(String fileString) {
```
Perhaps rename it like so? Note that methods have to be verbs and in camel case
```suggestion
    public static Event getEventFromFileString(String fileString) {
```

drustanyjt replied:
@SherisseTJW thanks for spotting these issues! Sorry I forgot to reply earlier, but I've changed the things you listed to align with the style guide.

SherisseTJW reviewed (commented):
Overall LGTM!

Just some minor code suggestions to follow the coding standards :3

drustanyjt committed:
add JavaFX to gradle and try simple chat interface

drustanyjt committed:
add full GUI and fix load from save file bug

drustanyjt committed:
add checkstyle fixes

drustanyjt committed:
Merge branch 'branch-A-CheckStyle'

drustanyjt committed:
Fix typo in image file name

The path to image files was capitalised incorrectly, which caused the
jar file from Shadow to encounter a null stream exception at runtime.

drustanyjt committed:
Add simple assertions in DukeApp

These are simple assertions to ensure no runtime
errors occur from uninstantiated components.

drustanyjt committed:
Improve code quality through refactoring

The main source of improvements was by moving the presentation logic
from `DukeApp` to `Ui`.

Previously, when responding to a user command, `DukeApp` would assume
that the responses would be preinted to `System.out` and redirect it
from there.

Now, `DukeApp` relies completely on `Ui` for displaying respones, and is
only responsible for executing commands.

Additionally, many minor changes were made in accordance with the Code
Style.

drustanyjt committed:
Merge branch 'master' into branch-A-CodeQuality

drustanyjt committed:
Merge pull request #1 from drustanyjt/branch-A-CodeQuality

Improve code quality

drustanyjt committed:
Add CI for prereleases and unit tests

drustanyjt committed:
Fix versioning in marketplace actions

drustanyjt committed:
Merge pull request #2 from drustanyjt/branch-A-CI

Add checkpoint A-CI

drustanyjt committed:
Remove release trigger for push to main

drustanyjt committed:
Fix workflow missing Gradle setup

drustanyjt committed:
Fix GitHub action not properly release jar files

drustanyjt committed:
Use latest checkout action in workflows

drustanyjt committed:
Use ref_name for release

drustanyjt committed:
Add the `stat` command to display stats

The current iteration of the `stat` command only displays a pie chart
of possible tasks broken down into their types. Future iterations may
see extensions being added to this.

To use the `stat` command simply type
```
stat
```
in the user input field.

This commit also adds resizing to the app window, and auto scroll.

drustanyjt committed:
Merge pull request #3 from drustanyjt/branch-BCD-Extension

Add BCD-Extension

drustanyjt committed:
Ignore CI tests on tag push

drustanyjt committed:
Rename instances of Duke to Fluffy

drustanyjt committed:
Add some more tests and refactor UI components

drustanyjt committed:
Add new documentation

drustanyjt committed:
Edit docs for clarity

drustanyjt committed:
Fix config in release yaml for github actions

drustanyjt committed:
Adjust JavaDocs and make Parser methods private

drustanyjt committed:
Fix spacing issue

drustanyjt committed:
Add blank check to parser

drustanyjt committed:
Update build-and-prerelease.yml

drustanyjt committed:
Merge pull request #6 from drustanyjt/drustanyjt-patch-1

Update build-and-prerelease.yml

drustanyjt committed:
Update build-and-prerelease.yml

drustanyjt committed:
Merge pull request #7 from drustanyjt/drustanyjt-patch-2

Update build-and-prerelease.yml

drustanyjt committed:
Update build-and-prerelease.yml

drustanyjt committed:
Merge pull request #8 from drustanyjt/drustanyjt-patch-3

Update build-and-prerelease.yml

drustanyjt committed:
Update build-and-prerelease.yml

drustanyjt committed:
Merge pull request #9 from drustanyjt/drustanyjt-patch-4

Update build-and-prerelease.yml

drustanyjt committed:
Update build-and-prerelease.yml

drustanyjt committed:
Merge pull request #10 from drustanyjt/drustanyjt-patch-5

Update build-and-prerelease.yml

drustanyjt committed:
Add nightly workflow

drustanyjt committed:
Merge branch 'master' of https://github.com/drustanyjt/ip
